portfolio/src/main/webapp/request.js

```diff
@@ -25,4 +25,109 @@ class ScheduleRequest {
   }
 }
 
-// TODO(tomasalvarez): write function to fetch results and display results
+// Once a call is made to the server for scheduling this array will be a local
+// copy of the results.
+var scheduledTasks = []
+
+    /**
+     * Gets all of the scheduling information from the UI and returns a
+     * ScheduleRequest with all of the data.
+     */
+    function
+    createRequestFromUiInformation() {
+      const startTime = document.getElementById('working-hour-start').value;
+      const endTime = document.getElementById('working-hour-end').value;
+      const events = collectAllEvents();
+      const tasks = collectAllTasks();
+      const algorithmType = document.getElementById('algorithm-type').value;
+      const scheduleRequest = new ScheduleRequest(
+          events, tasks, getTimeObject(startTime), getTimeObject(endTime),
+          algorithmType);
+      return scheduleRequest;
+    }
+
+/**
+ * Gets called when the user hits 'Start Scheduling'.
+ */
+function
+startScheduling() {
+  const scheduleRequest = createRequestFromUiInformation();
+  // Create the request to send to the server using the data we collected from
+  // the web form.
+  fetchScheduledTasksFromServlet(scheduleRequest).then((scheduledTaskArray) => {
+    updateResultsOnPage(scheduledTaskArray);
+  });
+}
+
+/**
+ * Updates the UI to show the results of a query.
+ */
+function updateResultsOnPage(scheduledTaskArray) {
+  const resultElement = document.getElementById('scheduled-task-list');
+  resultElement.innerHTML = '';
+  scheduledTaskArray.forEach(addScheduledTaskToPage, resultElement);
+}
+
+/**
+ * Handles sending the ScheduleRequest object to the servlet and returns the
+ * scheduled tasks.
+ */
+function
+fetchScheduledTasksFromServlet() {
```

Comment:
Again do we need a line break here? It seems that the comment above is longer than this line would be.

Reply:
Done.